File: Blog/src/main/java/com/tourism/blog/entity/Blog.java
package com.tourism.blog.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "blogs")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Blog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long authorId;
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;
    private LocalDateTime createdAt;

    @ElementCollection      //posebna tabela
    @CollectionTable(name = "blog_images", joinColumns = @JoinColumn(name = "blog_id"))
    @Column(name = "image_url")
    private List<String> imageUrls;
}

File: Frontend/src/types/blog.ts
export interface Blog {
  id: number;
  authorId: number;
  title: string;
  description: string;
  createdAt: string;
  imageUrls: string[];
  likeCount: number;
}

export interface CreateBlog {
  authorId: number;
  title: string;
  description: string;
  imageUrls: string[];
}

export interface Comment {
  id: number;
  blogId: number;
  authorId: number;
  authorUsername: string;
  text: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateComment {
  authorId: number;
  authorUsername: string;
  text: string;
}

export interface UpdateComment {
  text: string;
}